CMSgov/design-system
# [NO-TICKET] Clean up analytics code in Storybook

## Summary

Clean up our code for testing and mocking component analytics in Storybook to...
1. Make sure we don't have Storybook configured to automatically provide an `onAnalyticsEvent` prop to everything which muddies the waters and keeps the default analytics behavior from working in the design system
2. Keep the default functionality all the way to calling `window.utag.link` so we can be as close to production as possible, and now our mock for `window.utag.link` logs the Storybook Action instead of it being logged by circumventing the analyics code and not calling our mocked `utag.link` at all

## How to test

Try the HEAD of this branch as well as 6faf024d580182cddf0eb6811d5e3db5cceb574b and see if both analytics options work and if switching between them is bug-free.

1. Logging analytics (mocking) should both log to actions and to the console.
2. Using live analytics should send network events that start with "interact"
3. In testing both kinds of links (normal and new tab) you probably have to "persist logs" in dev tools, both for the console and for network
